### DISCUSSION TITLE
Ender 3 v2 with Sprite Pro and CR Touch

### DISCUSSION THREAD

**Discussion by xtremewebart** [mriscoc/Ender3V2S1#1133] [Q&A]
Hi there can someone provide me some working settings for this combination of hardware (Sprite Pro and CR Touch)
I am trying to configure the probe settings, physical settings, offsets etc

Thank you

**Comment by esrpradeep** [comment]
Hi, I did the same upgrade a few months back and it is pretty straightforward. 

In my case, the Probe settings were: (if the CR touch is mounted in the front housing - default position)
X: -30.5, Y:-41.5, Z: -1.02

However, since my bed is badly wrapped, I have used "Y offset Mod" to compensate the mesh points better.
Reference:  https://www.printables.com/model/444079-cr-touch-mount-for-sprite-extruder

My Ender 3 v2 with Sprite Pro and CR Touch:
![20231116_082628](https://github.com/mriscoc/Ender3V2S1/assets/65079442/02ac900d-2d59-4c12-af1d-a3d02c5ce560)
With Mod, the new settings are:
X: -49.3, Y:0, Z: -2.02

More importantly: Tune the E-Steps. 'Make use of Extrude 100 mm from the Menu->Prepare->Move Axis-> Move Ext: 100mm"
In my case, it is: 429.21

Physical setting: 
It should be the same as before. For safer reasons, I keep my Max X, Y and Z tuned to 220 mm.

In Probe Settings:
Turn OFF: HS mode for CR touch. This is because of a rare Issue observed preventing CR touch from probing properly during UBL mesh points with 9x9.

Additionally, I have used custom firmware to make use of the T13 Thermistor and experiment with "Linear Advance" using the special configurator.

Hope the above information helps set up your Ender 3 V2 hardware upgrade.  If you need any help in building custom firmware I would be happy to help you either with my notes or share a custom firmware that might suit your needs.

Happy 3D printing!

Cheers,
ESR@Pradeep

**Reply by CharlesNNNie** [reply]
I have the same setup. I have almost the same setting as yours without the Y offset mod (except the z offset setting) I found that z prob offset is linked with normal z offset, so I'm not sure if that will affect the auto mesh levelling result. Because I'm having some issues with the auto mesh levelling, especially on the top left corner. The nozzle is much higher when travelling to the top left corner area so the filament does not stick to the bed like other areas. 
I'm not sure if you have had this kind of issue?

**Reply by esrpradeep** [reply]
Hello Charles,

_"The nozzle is much higher when travelling to the top left corner area so the filament does not stick to the bed like other areas."_ 
I can relate to this issue, as I ran into the same problem in the bottom left corner of the print area.

A possible cause could be that the printing bed (heating bed/print bed) may not be perfectly flat, causing it to tilt slightly.
The other reason is that the placement of the probe, which measures the distance between the nozzle and the bed, may not be ideal, leading to inaccurate measurements. For instance, even though the probe reports a Z offset of 0.02 when it reaches the endpoints, the nozzle may actually encounter greater bed tilt due to the probe's placement.

**Solutions:**
Use a glass bed: A glass bed provides a flatter printing surface, reducing the impact of bed warping.
Apply the "Y offset mod": This modification adjusts the probe's offset in the Y direction, minimizing the effect of bed warping along the Y axis.

**Tips for Building a Mesh:**
1. **Start with Tramming Wizard:**  Use the Tramming Wizard with minimum limits to ensure initial bed leveling.
2. **Ensure probe and nozzle alignment:** Verify that the probe and nozzle points align correctly.
3. **Utilize tilt mesh:** Employ the tilt mesh feature with a maximum of three probe points to capture the bed's tilt.
4. **Fill up the mesh:** Perform a "fill up the mesh" operation to complete the mesh generation.
5. **Adjust Z offset:** Adjust the Z offset using the paper method for fine-tuning.
6. **Load mesh:** Use the appropriate G-code to load the generated mesh (UBL or normal) before print.
7. **Test bed calibration:** Print a bed calibration test to confirm the effectiveness of the adjustments.

References:
https://github.com/mriscoc/Ender3V2S1/wiki/Bed-tramming-menu 
https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch 

Happy Printing!

Cheers,
ESR@Pradeep

**Reply by CharlesNNNie** [reply]
Thanks for your reply. I'll try those methods you mentioned. Currently what I'm doing is manually editing the mesh and bringing the top left down a bit. Just can't run an auto mesh again to cover the edited mesh.
